File: week2/Polyglot/Ruby/program.rb
require "base64"

class Proc
  def self.compose(f, g)
    lambda { |*args| f[g[*args]] }
  end
  def *(g)
    Proc.compose(self, g)
  end
end


class RubyRubyRuby
   def initialize(what_to_decode)
      @what_to_decode = what_to_decode
   end
   def answer
      puts "You got that right!"
      puts "You should check out the answer and you might like Ruby!"
      once = lambda { |x| Base64.decode64(x) }
      twice = lambda { |x| Base64.decode64(x) }
      result = once * twice
      puts "Your answer is:"
      puts result[@what_to_decode]
   end
end

hello = RubyRubyRuby.new("YUhSMGNITTZMeTkzZDNjdWNuVmllUzFzWVc1bkxtOXlaeTlpWnk4PQo=\n")
hello.answer

File: week2/Polyglot/C++/program.cpp
#include "base64.h"
#include <iostream>

int main() {
  std::string encoded = "aGVhZGVyIGZpbGVz";
  std::string decoded = base64_decode(encoded);

  std::cout << "You got this right!" << std::endl;
  std::cout << "C++ requires you to compile the base64.cpp file as well, in order to have the functions" << std::endl;
  std::cout << "Your answer is:" << std::endl;
  std::cout << decoded << std::endl;

  return 0;
}
